File: app/src/main/java/weatherapp/tittojose/me/weatherapp/model/APIClient.java
package weatherapp.tittojose.me.weatherapp.model;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import static weatherapp.tittojose.me.weatherapp.model.Constants.BASE_URL;


/**
 * Created by titto.jose on 14-12-2017.
 */

public class APIClient {
    private static Retrofit retrofit = null;

    public static Retrofit getClient(String baseUrl) {
        if (retrofit == null) {

            HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
            OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();

            retrofit = new Retrofit.Builder()
                    .baseUrl(baseUrl)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(client)
                    .build();
        }
        return retrofit;
    }


    public static WeatherAPI getWeatherAPIClient() {
        return APIClient.getClient(BASE_URL).create(WeatherAPI.class);
    }
}

File: app/src/main/java/weatherapp/tittojose/me/weatherapp/model/APIClient.kt
package weatherapp.tittojose.me.weatherapp.model

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object APIClient {
	private const val BASE_URL = "http://api.weatherstack.com/"
	private fun getClient(): Retrofit {
		val interceptor = HttpLoggingInterceptor()
		interceptor.level = HttpLoggingInterceptor.Level.BODY
		val client = OkHttpClient.Builder().addInterceptor(interceptor).build()
		return Retrofit.Builder()
			.baseUrl(BASE_URL)
			.addConverterFactory(GsonConverterFactory.create())
//			.addCallAdapterFactory(RxJava2CallAdapterFactory.create())
			.client(client)
			.build()
	}

	val weatherAPIClient: WeatherAPI
		get() = getClient().create(WeatherAPI::class.java)
}